src/dbnode/storage/index.go

```diff
@@ -524,8 +539,23 @@ func (i *nsIndex) writeBatches(
 				batch.MarkUnmarkedEntryError(m3dberrors.ErrTooPast, idx)
 				return
 			}
+
+			if i.forwardIndexDice.forwardIndexing(entry.Timestamp) {
```

Comment:
once you fix the global rand usage, will need to be careful about using this dice

Reply:
if we already have a lock that we can use to protect the dice, than it's gucci. if not, will have to think about the contention it'll introduce to add a lock here.

Reply:
I'll see if there's a way to avoid having to get a lock here; there's a lock above batch processing I could `roll` in, but that would mean that the forward write would be applied to the whole batch. If that's fine, should be easy :+1: